clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1027,6 +1030,19 @@ static QualType calculateKernelNameType(ASTContext &Ctx,
   return TAL->get(0).getAsType().getCanonicalType();
 }
 
+// Kernel names are currently mangled as type names which
+// may collide (in the IR) with the "real" type names generated
+// for RTTI etc when compiling host and device code together.
+// Therefore the mangling of the kernel function is changed for
+// NativeCPU to avoid such potential collision.
+static void changeManglingForNativeCPU(std::string &Name) {
```

Comment:
@erichkeane can you take a look at this? I do not know the implications of this.

Reply:
Yeah you are right, we should just add a suffix to the function name as you are proposing. Another reason why we are making this changes to mangling is that we ran into some weird codegen/linking bugs in the past, which were fixed by these mangling change, but unfortunately I don't have a simple reproducer available - maybe they were related to the ABI break that you mention.

Reply:
I changed our mangling to just add the `_NativeCPUKernel` suffix as discussed